lib/ui/src/containers/nav.js

```diff
@@ -95,6 +96,64 @@ const createMenu = memoize(1)((api, shortcutKeys, isFullscreen, showPanel, showN
   },
 ]);
 
+const recursiveRemove = (stories, id) => {
+  const node = stories[id];
+  // eslint-disable-next-line no-param-reassign
+  delete stories[id];
+  if (node.parent) {
+    const parent = stories[node.parent];
+    parent.children = parent.children.filter(child => child !== id);
+    if (!parent.children.length) {
+      recursiveRemove(stories, parent.id);
+    }
+  }
+};
+
+export const removeDocsOnlyStories = stories => {
+  const filtered = cloneDeep(stories);
+  Object.values(stories).forEach(({ id, parameters }) => {
+    if (parameters && parameters.docsOnly) {
+      recursiveRemove(filtered, id);
+    }
+  });
+  return filtered;
+};
+
+export const flattenDocsOnlyStories = stories => {
+  const filtered = cloneDeep(stories);
+  Object.values(stories).forEach(story => {
+    const { id, isLeaf, parent, parameters } = story;
+    if (isLeaf && parameters && parameters.docsOnly) {
+      if (!filtered[parent].isLeaf) {
+        filtered[parent].isLeaf = true;
```

Comment:
In theory this has the same problem as `removeLeafStories`.

Also you could probably do a no-`cloneDeep` version very similar to below

Reply:
In fact a `removeLeaves(stories, predicate?)` function would make sense and remove the need for the duplicate logic.